File: generate-icons.js
#!/usr/bin/env node

/**
 * Icon Generator for Obsidian Read Later
 * Generates PNG icons in 16x16, 48x48, and 128x128 sizes
 * 
 * This script creates simple canvas-based icons without external dependencies.
 * The icon design features a book/document with a bookmark, representing "read later".
 */

const fs = require('fs');
const { createCanvas } = require('canvas');

// Icon design: A document/book with a bookmark
function drawIcon(ctx, size) {
  const scale = size / 128; // Base design on 128x128
  
  // Clear canvas
  ctx.clearRect(0, 0, size, size);
  
  // Background circle with gradient
  const gradient = ctx.createLinearGradient(0, 0, size, size);
  gradient.addColorStop(0, '#7C3AED'); // Purple
  gradient.addColorStop(1, '#5B21B6'); // Darker purple
  
  ctx.fillStyle = gradient;
  ctx.beginPath();
  ctx.arc(size / 2, size / 2, size / 2 - 2 * scale, 0, Math.PI * 2);
  ctx.fill();
  
  // Document/Book shape
  const docWidth = 50 * scale;
  const docHeight = 65 * scale;
  const docX = (size - docWidth) / 2;
  const docY = (size - docHeight) / 2 + 5 * scale;
  
  // Document background
  ctx.fillStyle = '#FFFFFF';
  ctx.fillRect(docX, docY, docWidth, docHeight);
  
  // Document lines (text representation)
  ctx.strokeStyle = '#9CA3AF';
  ctx.lineWidth = 2 * scale;
  
  const lineSpacing = 12 * scale;
  const lineStartX = docX + 8 * scale;
  const lineEndX = docX + docWidth - 8 * scale;
  
  for (let i = 0; i < 4; i++) {
    const y = docY + 15 * scale + i * lineSpacing;
    ctx.beginPath();
    ctx.moveTo(lineStartX, y);
    ctx.lineTo(lineEndX, y);
    ctx.stroke();
  }
  
  // Bookmark ribbon
  const bookmarkWidth = 12 * scale;
  const bookmarkX = docX + docWidth - bookmarkWidth - 5 * scale;
  const bookmarkY = docY - 5 * scale;
  const bookmarkHeight = 35 * scale;
  
  ctx.fillStyle = '#F59E0B'; // Amber color for bookmark
  ctx.beginPath();
  ctx.moveTo(bookmarkX, bookmarkY);
  ctx.lineTo(bookmarkX + bookmarkWidth, bookmarkY);
  ctx.lineTo(bookmarkX + bookmarkWidth, bookmarkY + bookmarkHeight);
  ctx.lineTo(bookmarkX + bookmarkWidth / 2, bookmarkY + bookmarkHeight - 5 * scale);
  ctx.lineTo(bookmarkX, bookmarkY + bookmarkHeight);
  ctx.closePath();
  ctx.fill();
}

async function generateIcons() {
  const sizes = [16, 48, 128];
  
  console.log('Generating icons for Obsidian Read Later...\n');
  
  for (const size of sizes) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Draw the icon
    drawIcon(ctx, size);
    
    // Save as PNG
    const buffer = canvas.toBuffer('image/png');
    const filename = `icons/icon${size}.png`;
    
    fs.writeFileSync(filename, buffer);
    console.log(`✓ Generated ${filename} (${size}x${size})`);
  }
  
  console.log('\n✓ All icons generated successfully!');
}

// Check if canvas module is available
try {
  require.resolve('canvas');
  generateIcons().catch(err => {
    console.error('Error generating icons:', err);
    process.exit(1);
  });
} catch (e) {
  console.error('Error: "canvas" module not found.');
  console.error('Please install it with: npm install canvas');
  console.error('\nAlternatively, you can use the fallback SVG method.');
  process.exit(1);
}

File: generate-icons.py
#!/usr/bin/env python3
"""
Icon Generator for Obsidian Read Later
Generates PNG icons in 16x16, 48x48, and 128x128 sizes using PIL/Pillow
"""

try:
    from PIL import Image, ImageDraw
    import os
except ImportError:
    print("Error: Pillow library not found.")
    print("Please install it with: pip install Pillow")
    exit(1)

def draw_icon(size):
    """Draw the Obsidian Read Later icon"""
    # Create image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    scale = size / 128
    
    # Background circle with purple gradient (simplified as solid color)
    padding = int(2 * scale)
    circle_radius = size // 2 - padding
    
    # Draw purple circle background
    draw.ellipse(
        [padding, padding, size - padding, size - padding],
        fill=(124, 58, 237, 255),  # Purple #7C3AED
        outline=None
    )
    
    # Document/Book shape
    doc_width = int(50 * scale)
    doc_height = int(65 * scale)
    doc_x = (size - doc_width) // 2
    doc_y = (size - doc_height) // 2 + int(5 * scale)
    
    # Document background (white)
    draw.rectangle(
        [doc_x, doc_y, doc_x + doc_width, doc_y + doc_height],
        fill=(255, 255, 255, 255),
        outline=None
    )
    
    # Document lines (text representation)
    line_color = (156, 163, 175, 255)  # Gray #9CA3AF
    line_width = max(1, int(2 * scale))
    line_spacing = int(12 * scale)
    line_start_x = doc_x + int(8 * scale)
    line_end_x = doc_x + doc_width - int(8 * scale)
    
    for i in range(4):
        y = doc_y + int(15 * scale) + i * line_spacing
        draw.line(
            [(line_start_x, y), (line_end_x, y)],
            fill=line_color,
            width=line_width
        )
    
    # Bookmark ribbon (amber color)
    bookmark_width = int(12 * scale)
    bookmark_x = doc_x + doc_width - bookmark_width - int(5 * scale)
    bookmark_y = doc_y - int(5 * scale)
    bookmark_height = int(35 * scale)
    
    # Draw bookmark as a polygon
    bookmark_points = [
        (bookmark_x, bookmark_y),
        (bookmark_x + bookmark_width, bookmark_y),
        (bookmark_x + bookmark_width, bookmark_y + bookmark_height),
        (bookmark_x + bookmark_width // 2, bookmark_y + bookmark_height - int(5 * scale)),
        (bookmark_x, bookmark_y + bookmark_height)
    ]
    
    draw.polygon(
        bookmark_points,
        fill=(245, 158, 11, 255),  # Amber #F59E0B
        outline=None
    )
    
    return img

def generate_icons():
    """Generate all icon sizes"""
    sizes = [16, 48, 128]
    
    # Ensure icons directory exists
    os.makedirs('icons', exist_ok=True)
    
    print('Generating icons for Obsidian Read Later...\n')
    
    for size in sizes:
        img = draw_icon(size)
        filename = f'icons/icon{size}.png'
        img.save(filename, 'PNG')
        print(f'✓ Generated {filename} ({size}x{size})')
    
    print('\n✓ All icons generated successfully!')

if __name__ == '__main__':
    generate_icons()
